packages/atomic/src/components/insight/atomic-insight-facet/atomic-insight-facet.tsx

```diff
@@ -0,0 +1,277 @@
+ * @part search-clear-button - The button to clear the search box of input.
+ * @part more-matches - The label indicating there are more matches for the current facet search query.
+ * @part no-matches - The label indicating there are no matches for the current facet search query.
+ * @part matches-query - The highlighted query inside the matches labels.
+ * @part search-highlight - The highlighted query inside the facet values.
+ *
+ * @part values - The facet values container.
+ * @part value-label - The facet value label, common for all displays.
+ * @part value-count - The facet value count, common for all displays.
+ *
+ * @part value-checkbox - The facet value checkbox, available when display is 'checkbox'.
+ * @part value-checkbox-checked - The checked facet value checkbox, available when display is 'checkbox'.
+ * @part value-checkbox-label - The facet value checkbox clickable label, available when display is 'checkbox'.
+ * @part value-link - The facet value when display is 'link'.
+ * @part value-link-selected - The selected facet value when display is 'link'.
+ * @part value-box - The facet value when display is 'box'.
+ * @part value-box-selected - The selected facet value when display is 'box'.
+ *
+ * @part show-more - The show more results button.
+ * @part show-less - The show less results button.
+ * @part show-more-less-icon - The icons of the show more & show less buttons.
+ */
+@Component({
+  tag: 'atomic-insight-facet',
+  styleUrl: 'atomic-insight-facet.pcss',
+  shadow: true,
+})
+export class AtomicInsightFacet
+  implements InitializableComponent<InsightBindings>, BaseFacet<InsightFacet>
+{
+  @InitializeBindings() public bindings!: InsightBindings;
+  public facetCommon!: FacetCommon;
+  public facet!: InsightFacet;
+  public searchStatus!: InsightSearchStatus;
+  @Element() private host!: HTMLElement;
+
+  @BindStateToController('facet')
+  @State()
+  public facetState!: InsightFacetState;
+  @BindStateToController('searchStatus')
+  @State()
+  public searchStatusState!: InsightSearchStatusState;
+  @State() public error!: Error;
+
+  /**
+   * Specifies a unique identifier for the facet.
+   */
+  @Prop({mutable: true, reflect: true}) public facetId?: string;
+  /**
+   * The non-localized label for the facet.
+   * Used in the `atomic-breadbox` component through the bindings store.
+   */
+  @Prop({reflect: true}) public label = 'no-label';
+  /**
+   * The field whose values you want to display in the facet.
+   */
+  @Prop({reflect: true}) public field!: string;
+  /**
+   * The number of values to request for this facet.
+   * Also determines the number of additional values to request each time more values are shown.
+   */
+  @Prop({reflect: true}) public numberOfValues = 8;
+  /**
+   * Whether this facet should contain a search box.
+   * When "true", the search is only enabled when more facet values are available.
+   */
+  @Prop({reflect: true}) public withSearch = true;
```

Comment:
From what I remember from @lvu285 comment yesterday we could remove the facet search feature (ie: always set it to false in the headless controller)

Reply:
👍  For the parts in the doc too

Reply:
they have not yet supported the search for standard facet for 1st release, maybe the future yes. So if it's too complicated to make it works now, we can remove it, If you can make it works without putting much effort on it, then keep it